### PR TITLE
Backup status and restic image

### PR DESCRIPTION
**Pull request by gsanchietti**
Add backup status info:
- total size
- number of files
- start date
- end date
- success status

Also use a custom redis image.

### CODE REVIEW

File: core/imageroot/var/lib/nethserver/cluster/actions/list-backups/50list [hunk 1]
```diff
@@ -49,11 +49,17 @@ for mbackup in rdb.scan_iter('module/*/backups'):
         if bid in backups:
             mid = mbackup.removeprefix('module/').removesuffix('/backups')
             module_uuid = rdb.hget(f"module/{mid}/environment", "MODULE_UUID")
+            status = rdb.hgetall(f"module/{mid}/backup_status/backup{bid}")
```
Comment (DavidePrincipi) [inline core/imageroot/var/lib/nethserver/cluster/actions/list-backups/50list:52]:
```suggestion
            status = rdb.hgetall(f"module/{mid}/backup_status/{bid}")
```

File: core/imageroot/usr/local/agent/pypkg/agent/backup/__init__.py [hunk 2]
```diff
@@ -33,6 +35,7 @@ class Restic:
     rootfull = False
     restic_dir = ""
     restic_env = ""
+    restic_image = "ghcr.io/nethserver/restic"
```
Comment (DavidePrincipi) [inline core/imageroot/usr/local/agent/pypkg/agent/backup/__init__.py:38]: The value should be read from core.env. For example:

```python
import agent
core_env = agent.read_envfile('/etc/nethserver/core.env')
print(core_env["RESTIC_IMAGE"])
```

File: core/build-image.sh [hunk 3]
```diff
@@ -107,6 +108,16 @@ buildah commit "${container}" "${repobase}/${reponame}"
 buildah rm "${container}"
 images+=("${repobase}/${reponame}")
 
+echo "Building the restic image..."
+container=$(buildah from alpine)
+reponame="restic"
+buildah run ${container} -- apk add restic
```
Comment (DavidePrincipi) [inline core/build-image.sh:114]:
```suggestion
buildah run ${container} -- apk add --no-cache restic
```

...or add cache index cleanup

```
f0657440c06d:/# du -h -s /var/cache/apk/
2.2M	/var/cache/apk/
```

Comment (gsanchietti) [reply core/build-image.sh:114]: Cache is disabled inside the image

File: core/imageroot/usr/local/agent/pypkg/agent/backup/__init__.py (hunk 2 above)
Comment (gsanchietti) [reply core/imageroot/usr/local/agent/pypkg/agent/backup/__init__.py:38]: I prefer to leave it inside the library

File: core/build-image.sh (hunk 3 above)
Comment (DavidePrincipi) [reply core/build-image.sh:114]: 2.2M of index files are retained anyway...

File: core/imageroot/usr/local/agent/pypkg/agent/backup/__init__.py (hunk 2 above)
Comment (DavidePrincipi) [reply core/imageroot/usr/local/agent/pypkg/agent/backup/__init__.py:38]: So, why did you add it to core.env? An hardcoded value would be enough.

Setting the value in core.env allows to run `podman pull $RESTIC_IMAGE` during modules installation (and just before the first backup run).

Comment (gsanchietti) [reply core/imageroot/usr/local/agent/pypkg/agent/backup/__init__.py:38]: Because you asked to add it to the env :)

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**